Physics/Dynamics/ProductionFormFactor.hpp

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2013, 2019 The ComPWA Team.
+// This file is part of the ComPWA framework, check
+// https://github.com/ComPWA/ComPWA/license.txt for details.
+
+#ifndef PHYSICS_DYNAMICS_PRODUCTION_FORMFACTOR_HPP_
+#define PHYSICS_DYNAMICS_PRODUCTION_FORMFACTOR_HPP_
+
+#include <boost/property_tree/ptree.hpp>
+#include <memory>
+#include <vector>
+
+#include "AbstractDynamicalFunction.hpp"
+#include "Core/Exceptions.hpp"
+#include "Core/Functions.hpp"
+#include "Core/Spin.hpp"
+#include "FormFactor.hpp"
+
+namespace ComPWA {
+namespace Physics {
+namespace Dynamics {
+
+/// The production formfactor is implemented in Blatt-Weisskopf type \f$B_{L}(q)\f$
+/// Ref. Dalitz Plot Analysis, PDG 2012, where 
+/// \f[
+/// \frac{B^{\prime}(q_{0})}{B^{\prime}_{L}(q)} = q^{L}B^{\prime}_{L}(q, q_{0})
+/// \f]
+class ProductionFormFactor
+    : public ComPWA::Physics::Dynamics::AbstractDynamicalFunction {
+
+public:
+  //============ CONSTRUCTION ==================
+  ProductionFormFactor(std::string name,
```

Comment:
Try to make the constructor get all of the variables it need to be fully constructed. In this case a L, a pair of masses, meson radius, form factor type. All of the xml extraction should go into the XMLBuilder

Reply:
Btw I think its good if we introduce a new XML tag for the form factors `<FormFactor>`, which contains all of the necessary information

Reply:
OK. That is good idea, I will try it.